Src/AutoFixtureUnitTest/FreezeOnMatchCustomizationTest.cs

```diff
@@ -77,6 +90,63 @@ public void InitializeWithMatcherShouldSetCorrespondingProperty()
             Assert.Equal(matcher, sut.Matcher);
         }
 
+
+        [Fact]
+        public void InitializeWithSingleNullRequestArgumentShouldThrowArgumentNullException()
+        {
+            Assert.Throws<ArgumentNullException>(() =>
+                new FreezeOnMatchCustomization((object)null));
+        }
+
+        [Fact]
+        public void InitializeWithNullRequestShouldThrowArgumentNullException()
+        {
+            var ex = Assert.Throws<ArgumentNullException>(() =>
+                new FreezeOnMatchCustomization((object)null, new FalseRequestSpecification()));
+            Assert.Equal("request", ex.ParamName);
+        }
+
+        [Fact]
+        public void InitializeWithRequestAndNullMatcherShouldThrowArgumentNullException()
+        {
+            var ex = Assert.Throws<ArgumentNullException>(() =>
+                  new FreezeOnMatchCustomization(new object(), null));
+            Assert.Equal("matcher", ex.ParamName);
+        }
+
+        [Fact]
+        public void InitializeWithSingleRequestShouldSetCorrespondingProperty()
+        {
+            var request = new object();
+            var sut = new FreezeOnMatchCustomization(request);
+            Assert.Same(request, sut.Request);
+        }
+
+        [Fact]
+        public void InitializeWithObjectRequestShouldSetTrueRequestSpecificationMatcher()
```

Comment:
@sergeyshushlyapin Yes, if no other test fails (and it should be so), I'd say that it's better to make the code even more simple. I wouldn't consider that as a breaking change, because we expose `IRequestSpecification` rather than a concrete type, so it's fine to change the underlying implementation.